src/geometry/WorkingGeometry.ts

```diff
@@ -38,4 +39,36 @@ export class WorkingGeometry {
      * must be equal to 0 to ensure that it is a point in Affine space.
      */
     public controlPoints: vec4[];
+
+    /**
+     * Return the BakedGeometry representing the current state of the WorkingGeomtry.
+     *
+     * @returns {BakedGeometry}
+     */
+    public bake(): BakedGeometry {
+        const bakedVertices = this.vertices.map((workingVec: vec4) =>
+            vec3.fromValues(workingVec[0], workingVec[1], workingVec[2])
+        );
+        const bakedIndecies = this.faces.reduce((accum: number[], face: Face) => {
+            return accum.concat(face.indices);
+        }, []);
+        const bakedNormals = this.faces.map((face: Face) => {
+            return vec3.fromValues(face.normal[0], face.normal[1], face.normal[2]);
+        });
+
+        // Make all of the baked shapes red for now.
+        const color = vec3.fromValues(255, 0, 0);
+        const bakedColors: vec3[] = [];
```

Comment:
Could you write this as:

```typescript
const bakedColors: vec3[] = bakedVertices.map(() => vec3.fromValues(1, 0, 0));
```

Thank you! 🙂 

___

Btw, the colour is changed from `255, 0, 0` to `1, 0, 0` because regl uses float values in the range 0 to 1.